packages/core/src/indexing/postgres/__tests__/postgres-index-api.test.ts

```diff
@@ -147,17 +161,52 @@ describe('indexStream', () => {
   test('new stream', async () => {
     const now = new Date()
     await indexApi.indexStream(STREAM_CONTENT)
-    const result: Array<any> = await dbConnection.from(`${MODELS_TO_INDEX[0]}`).select('*')
+    let result: Array<any> = await dbConnection
+      .from(`${MODELS_TO_INDEX[0]}`)
+      .select(
+        dbConnection.raw(
+          `*, stream_content->'settings'->'dark_mode' AS dark_mode, stream_content->>'id' AS id`
+        )
+      )
     expect(result.length).toEqual(1)
     const raw = result[0]
     expect(raw.stream_id).toEqual(STREAM_ID_B)
     expect(raw.controller_did).toEqual(CONTROLLER)
+    expect(raw.stream_content).toEqual(STREAM_TEST_DATA_PROFILE)
+    expect(raw.tip).toEqual(FAKE_CID.toString())
     expect(raw.last_anchored_at).toBeNull()
     expect(raw.first_anchored_at).toBeNull()
     const createdAt = new Date(raw.created_at)
     const updatedAt = new Date(raw.updated_at)
     expect(closeDates(createdAt, now)).toBeTruthy()
     expect(closeDates(updatedAt, now)).toBeTruthy()
+    expect(raw.dark_mode).toEqual(STREAM_TEST_DATA_PROFILE.settings.dark_mode)
+    expect(raw.id).toEqual(STREAM_TEST_DATA_PROFILE.id)
+
+    // create index on jsonb content
+    result = await dbConnection.raw(
+      `CREATE INDEX idx_postgres_jsonb ON ${MODELS_TO_INDEX[0]}(stream_id, (stream_content->'settings'->'dark_mode'))`
+    )
+    expect(result.command).toEqual('CREATE')
+
+    // verify index usage
+    await dbConnection.raw(`SET enable_seqscan = off;`)
+    result = await dbConnection.raw(
+      `EXPLAIN SELECT *
+       FROM ${MODELS_TO_INDEX[0]}
+       WHERE stream_id='${STREAM_ID_B}'
```

Comment:
querying on the streamid seems to negate querying on the content since there will only ever be one match on streamid anyway.  Can we query *just* on the content?

Reply:
ooh, yes I thought wrongly inserts vs. updates (override) here. I'll adjust and do just the content.